orchagent/portsorch.cpp

```diff
@@ -4905,6 +4922,88 @@ bool PortsOrch::getSystemPorts()
     return true;
 }
 
+bool PortsOrch::addInbandVlan(const string &alias, const string &type)
+{
+    if (type != "vlan")
+    {
+        SWSS_LOG_WARN("Inband type %s is not vlan.", type.c_str());
+        return false;
+    }
+    
+    /* Create inband Vlan and also local and remote CPU ports to inband Vlan */
+
+    Port cpuPort;
+    Port inbandVlan;
+    string tagged_mode("tagged");
+
+    if (!addVlan(alias))
+    {
+        SWSS_LOG_ERROR("Failed to add inband VLAN %s", alias.c_str());
+        return false;
+    }
+
+    if (!getPort(alias, inbandVlan))
+    {
+        SWSS_LOG_ERROR("Failed to get port object of inband VLAN %s", alias.c_str());
+        return false;
+    }
+
+    getCpuPort(cpuPort);
+
+    if (!addBridgePort(cpuPort))
+    {
+        SWSS_LOG_ERROR("Failed to add CPU port %s as bridge port", cpuPort.m_alias.c_str());
+        return false;
+    }
+
+    if (!addVlanMember(inbandVlan, cpuPort, tagged_mode))
+    {
+        SWSS_LOG_ERROR("Failed to add CPU port %s in inband VLAN", cpuPort.m_alias.c_str());
+        return false;
+    }
+
+    SWSS_LOG_NOTICE("add port %s to inbandVlan %s", cpuPort.m_alias.c_str(), inbandVlan.m_alias.c_str());
+
+    if (!addHostIntfs(inbandVlan, inbandVlan.m_alias, inbandVlan.m_hif_id))
+    {
+        SWSS_LOG_ERROR("Failed to add host intf for inband VLAN");
+        return false;
+    }
+
+    if (!setHostIntfsOperStatus(inbandVlan, true))
+    {
+        SWSS_LOG_ERROR("Failed to set host intf oper status for inband VLAN");
+        return false;
+    }
+
+    for (auto &it: m_portList)
+    {
+        Port port = it.second;
+        if (port.m_type != Port::SYSTEM ||
+            port.m_system_port_info.type == SAI_SYSTEM_PORT_TYPE_LOCAL ||
```

Comment:
can local cpu port be covered in this same loop ? basically this will make code common w.r.t API addBridgePort() and addVlanMemeber()

Reply:
This is a good point. I will try to see if I can move local cpu port processing into the for loop.

Reply:
Fixed as what you suggested.